tests/rptest/scale_tests/log_storage_target_size_test.py

```diff
@@ -0,0 +1,167 @@
+            self.manifest_upload_interval,
+            'log_storage_max_usage_interval':
+            self.log_storage_max_usage_interval,
+            'log_storage_target_size': target_size,
+        }
+        si_settings = SISettings(test_context=self.test_context,
+                                 log_segment_size=log_segment_size)
+        self.redpanda.set_extra_rp_conf(extra_rp_conf)
+        self.redpanda.set_si_settings(si_settings)
+        self.redpanda.start()
+
+        # Sanity check test parameters against the nodes we are running on
+        disk_space_required = data_size
+        assert self.redpanda.get_node_disk_free(
+        ) >= disk_space_required, f"Need at least {disk_space_required} bytes space"
+
+        # create the target topic
+        rpk = RpkTool(self.redpanda)
+        rpk.create_topic(topic_name, partitions=partition_count, replicas=3)
+
+        # setup and start the background producer
+        producer = KgoVerifierProducer(self.test_context,
+                                       self.redpanda,
+                                       topic_name,
+                                       msg_size=msg_size,
+                                       msg_count=msg_count,
+                                       rate_limit_bps=rate_limit_bps,
+                                       batch_max_bytes=msg_size * 8)
+        producer.start()
+        produce_start_time = time.time()
+
+        # helper: bytes to MBs / human readable
+        def hmb(bs):
+            convert = lambda b: round(b / (1024 * 1024), 1)
+            if isinstance(bs, int):
+                return convert(bs)
+            return [convert(b) for b in bs]
+
+        # while the producer is running, periodically wake up and go see if the
+        # kafka storage on every node is meeting the target, subject to the max
+        # overage fuzz factor.
+        check_interval = 5
+        last_report_time = 0
+        max_totals = None
+        while producer.produce_status.acked < msg_count:
+            # calculate and report disk usage across nodes
+            if time.time() - last_report_time >= check_interval:
+                totals = self._kafka_usage()
+                if not max_totals:
+                    max_totals = totals
+                max_totals = [max(t, m) for t, m in zip(totals, max_totals)]
+                overages = [t - target_size for t in totals]
+                violation = any(o > max_overage for o in overages)
+                self.logger.info(
+                    f"Acked messages {producer.produce_status.acked} / {msg_count}"
+                )
+                self.logger.info(
+                    f"Latest totals {hmb(totals)} maxes {hmb(max_totals)}")
+                self.logger.info(
+                    f"Target {hmb(target_size)} overages {hmb(overages)} max {hmb(max_overage)}"
+                )
+                assert not violation, f"Overage exceeded max {hmb(max_overage)}"
```

Comment:
nit: would be nice to print out on which nodes the violation occurred.

Reply:
Yeh, good point. I made this change locally will include it in a next PR. I think if I failure happened before that it could be inferred from the ordering of the lists printed here, but sorta annoying!